Framework/Algorithms/src/EstimateResolutionDiffraction.cpp

```diff
@@ -187,7 +190,12 @@ void EstimateResolutionDiffraction::estimateDetectorResolution() {
         spectrumInfo.isMonitor(i) ? 0.0 : spectrumInfo.twoTheta(i);
     double theta = 0.5 * twotheta;
 
-    double solidangle = det.solidAngle(samplepos);
+    double solidangle = 0.0;
+    for (const auto detID : m_inputWS->getSpectrum(i).getDetectorIDs()) {
+      const auto index = detectorInfo.indexOf(detID);
+      if (!detectorInfo.isMasked(index))
+        solidangle += detectorInfo.detector(index).solidAngle(samplepos);
```

Comment:
No, I did not get feedback. However, in I think all cases I refactored it did actually seem to make sense to ignore masked detectors. This is now done explicitly and manually by clients and thus more obvious to the developer, which I see as an advantage.

Regarding adding this as functionality to `DetectorInfo`, I agree, but I would avoid that at this point for two reasons:
- We are in the middle of some rather complex refactoring.
- I am not sure of this is something that should actually be part of the class interface. Maybe a stand alone method that just uses `DetectorInfo` might make more sense. Since this is not an issue that blocks us anywhere I would rather postpone this and similar design decision until the basics are more settled.